src/main/java/org/jabref/gui/preferences/websearch/WebSearchTabViewModel.java

```diff
@@ -138,7 +141,15 @@ public void setValues() {
         grobidEnabledProperty.setValue(grobidPreferences.isGrobidEnabled());
         grobidURLProperty.setValue(grobidPreferences.getGrobidURL());
 
-        apiKeys.setValue(FXCollections.observableArrayList(preferences.getImporterPreferences().getApiKeys()));
+        // Initialize tempApiKeys with a deep copy of the actual API Keys, so changes in the UI do not directly affect the actual settings
+        tempApiKeys.set(FXCollections.observableArrayList(
+                preferences.getImporterPreferences().getApiKeys().stream()
+                           .map(apiKey -> new FetcherApiKey(apiKey.getName(), apiKey.shouldUse(), apiKey.getKey()))
+                           .collect(Collectors.toList())
+        ));
+        // Set the UI-bound apiKeys property to tempApiKeys
+        apiKeys.set(tempApiKeys);
+
```

Comment:
You can do this directly on the `apiKeys` list without the need for `tempApiKeys`. 

Also, change the `apiKeys` list to an `ObservableList` instead of a `ListProperty`.

Reply:
Thank you for the feedback. I have simplified the code by directly working with the `apiKeys`.

I have also changed `apiKeys` from a `ListProperty` to an `ObservableList`  and deleted redundant code about saving perfermance.

I have edited and updated PR Description above. Thank you!